### PR TITLE
Add NFS export policy name to file system space metrics

### LINKED ISSUE #75: add export policy name to file system space metric(s)
File systems that use an export policy have the export policy data returned similar to rules in the initial get /filesystems.  The name of the policy can be used for owner/association of said file system and would be helpful in the metric return.

Currently, the filesystem for NFS looks like this in the exporter:

```
type Nfs struct {
	V3Enabled  bool   `json:"v3_enabled"`
	V41Enabled bool   `json:"v4_1_enabled"`
	Rules      string `json:"rules"`
}
``` 

However, there is an additional dict/None returned for Export policy in the nfs key response:
```
"nfs": {
    "v3_enabled": true,
    "v4_1_enabled": true,
    "rules": "",
    "export_policy": {
        "id": "string",
        "name": "string",
        "resource_type": "string"
}
```

Can we implement something like this(expand the above struct) and then add the name to the output metric schema(provisioned space example below)?
```
	if fs.Nfs.ExportPolicy.Name != "" {
		exppol = fs.Nfs.ExportPolicy.Name
	} else {
		exppol = "None"
	}
	ch <- prometheus.MustNewConstMetric(
		c.SpaceDesc,
		prometheus.GaugeValue,
		float64(fs.Provisioned),
		fs.Name, exppol, nfs, smb, "provisioned",
	)
```

### PR DESCRIPTION
**Pull request by chrroberts-pure**
closes #75 

FS with both smb and nfs with a policy
````
purefb_file_systems_space_bytes{name="testfs",nfs="3,41",nfspolicy="All_Access",smb="3",space="provisioned"} 4.294967296e+11
purefb_file_systems_space_bytes{name="testfs",nfs="3,41",nfspolicy="All_Access",smb="3",space="snapshots"} 0
purefb_file_systems_space_bytes{name="testfs",nfs="3,41",nfspolicy="All_Access",smb="3",space="total_physical"} 2.63722799297e+11
purefb_file_systems_space_bytes{name="testfs",nfs="3,41",nfspolicy="All_Access",smb="3",space="unique"} 2.63722799297e+11
purefb_file_systems_space_bytes{name="testfs",nfs="3,41",nfspolicy="All_Access",smb="3",space="virtual"} 3.49897389568e+11
````

only smb
````
purefb_file_systems_space_bytes{name="SMB_TEST",nfs="",nfspolicy="",smb="3",space="provisioned"} 0
purefb_file_systems_space_bytes{name="SMB_TEST",nfs="",nfspolicy="",smb="3",space="snapshots"} 0
purefb_file_systems_space_bytes{name="SMB_TEST",nfs="",nfspolicy="",smb="3",space="total_physical"} 0
purefb_file_systems_space_bytes{name="SMB_TEST",nfs="",nfspolicy="",smb="3",space="unique"} 0
purefb_file_systems_space_bytes{name="SMB_TEST",nfs="",nfspolicy="",smb="3",space="virtual"} 0
````

only nfs3
````
purefb_file_systems_space_bytes{name="RsyncTest",nfs="3",nfspolicy="",smb="",space="provisioned"} 1.099511627776e+12
purefb_file_systems_space_bytes{name="RsyncTest",nfs="3",nfspolicy="",smb="",space="snapshots"} 0
purefb_file_systems_space_bytes{name="RsyncTest",nfs="3",nfspolicy="",smb="",space="total_physical"} 0
purefb_file_systems_space_bytes{name="RsyncTest",nfs="3",nfspolicy="",smb="",space="unique"} 0
purefb_file_systems_space_bytes{name="RsyncTest",nfs="3",nfspolicy="",smb="",space="virtual"} 0
````

### CODE REVIEW

**Comment by chrroberts-pure** [comment]
@mboeckelmann please review at your leisure.

**Review by sdodsley — COMMENTED** [review]
Doesn't the spec need updating as well

**Comment by chrroberts-pure** [comment]
> Doesn't the spec need updating as well

Yes, a lot of the spec needs to be filled in - this will stay in draft until it’s ready for merge.

**Review by sdodsley — DISMISSED** [review]
Seems OK

**Comment by chrroberts-pure** [comment]
approved for merge

**Comment by sdodsley** [comment]
@chrroberts-pure this has a merge conflict

File: specification/metrics/purefb-metrics.md [hunk 1]
```diff
@@ -98,6 +98,8 @@ This document describes the semantic conventions for Pure FlashBlade Metrics.
...
+|                                                                                                                                                                                                                                    |                                                        |                                                                    |                  |                                                                                                                         |            | `space`              | `snapshots`, `total_physical`, `unique`, `virutal`, `object_count`                                                 |
```
Comment (chrroberts-pure) [inline specification/metrics/purefb-metrics.md:101]:
```suggestion
```

File: specification/metrics/purefb-metrics.md [hunk 2]
```diff
@@ -98,6 +98,8 @@ This document describes the semantic conventions for Pure FlashBlade Metrics.
...
+| Available                                                                                                                                                                                                                          | purefb_buckets_space_data_reduction_ratio              | FlashBlade buckets space data reduction ratio                      | ratio            | Gauge                                                                                                                   | Double     | `name`               | (name)                                                                                                             |
```
Comment (chrroberts-pure) [inline specification/metrics/purefb-metrics.md:102]:
```suggestion
```

File: specification/metrics/purefb-metrics.md (hunk 1 above)
Comment (chrroberts-pure) [inline specification/metrics/purefb-metrics.md:101]:
```suggestion
```

**Comment by chrroberts-pure** [comment]
the best ones always do - ready for merge @sdodsley

**Pull request merged**